lib/private/share20/defaultshareprovider.php

```
@@ -218,14 +218,14 @@ private function createShare($data) {
 		$share->setSharedBy($this->userManager->get($data['uid_owner']));
 
 		// TODO: getById can return an array. How to handle this properly??
-		$path = $this->userFolder->getById((int)$data['file_source']);
-		$path = $path[0];
-		$share->setPath($path);
+		$folder = $this->rootFolder->getUserFolder($share->getSharedBy()->getUID());
+		$path = $folder->getById((int)$data['file_source'])[0];
 
-		$owner = $path->getStorage()->getOwner('.');
-		if ($owner !== false) {
-			$share->setShareOwner($this->userManager->get($owner));
-		}
+		$owner = $path->getOwner();
+		$share->setShareOwner($owner);
+
+		$path = $this->rootFolder->getUserFolder($owner->getUID())->getById((int)$data['file_source'])[0];
```

yes... I need to think about that.. basically if the array is empty there is a data inconsitency... and maybe just throw an exception that explictly says that.